library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -150,9 +150,14 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
      */
     @RealZoom
     override val realZoom: Float
-        @RealZoom
         get() = zoom * mBaseZoom
 
+    /**
+     * The current pan as an [AbsolutePoint]
+     */
+    private val pan: AbsolutePoint
+        get() = AbsolutePoint(panX, panY)
```

Comment:
Are these pan and scaledPan values accessed frequently?
If they are it might be worth to avoid all these new instances by caching the value (have a private _pan and use _pan.set() in this getter), since touch events are a lot.

By the way I thought from another comment that these two were public?

Reply:
I made them public now, don't know why they weren't.
Yes using the same Object is probably better for performance - I changed that.

Reply:
I didn't know you could do what you did :-) just one last change, can they be vals of ZoomApi overriden in the engine?

Reply:
Well the `pan` val can, the `scaledPan` depends on `mTransformedRect` so I don't think so?